docs/design/schedule_algorithm.md

```diff
@@ -0,0 +1,39 @@
+
+# Schedule Algorithm in Kubernetes
+
+For each unscheduled Pod, the Kubernetes scheduler tries to find a node across the cluster according to a set of rules. A general introduction to the Kubernetes scheduler can be found at [docs/design/scheduler.md](../../docs/design/scheduler.md). In this document, the algorithm of how to select a node for the Pod is explained. There are two steps before a destination node of a Pod is chosen. The first step is filtering all the nodes and the second is ranking the remaining nodes to find a best fit for the Pod.
+
+## Filtering the nodes
+The purpose of filtering the nodes is to fitler out the nodes that do not meet certain requirements of the Pod. For example, if the free resource on a node (measured by the capacity minus the sum of the resource limits of all the Pods that already run on the node) is less than the Pod's required resource, the node should not be considered in the ranking phase so it is filtered out. Currently, there are several "predicates" implementing different filtering policies, including:
+
+- `NoDiskConflict`: Check if the free disk space meets the requirement of the Pod.
```

Comment:
This description is not correct. See description here: https://github.com/GoogleCloudPlatform/kubernetes/blob/c7701205bdbba219972dbba3ce25f6f98e257dce/plugin/pkg/scheduler/algorithm/predicates/predicates.go#L82